File: 卡码/代码随想录/Go/二叉树/02-二叉树属性/LeetCode_0559.go
//559. N 叉树的最大深度
package main

type Node struct {
	Val int
	Children []*Node
}

func maxDepth(root *Node) int {
	if root==nil {
		return 0
	}
	var depth int=0
	for _,node:=range root.Children {
		depth=max(depth,maxDepth(node))
	}
	return 1+depth
}

File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/二叉树/_02_二叉树属性/LeetCode_0559.java
package com.kurumi.codeThoughts.二叉树._02_二叉树属性;

import com.kurumi.codeThoughts.entity.Node1;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 559. N 叉树的最大深度
 */
public class LeetCode_0559 {
    static class Solution {
        public int maxDepth(Node1 root) {
            if (root == null)
                return 0;
            int depth = 0;
            for (Node1 node : root.children) {
                depth = Integer.max(depth, maxDepth(node));
            }
            return 1 + depth;
        }
    }
}


